### PR TITLE
feat: add 404 page to artwork

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-678] 

### Description
This PR adds a 404 page when an artwork screen fails at loading.

**Note:** This PR makes me think, do we really need to do this only here? maybe our error boundary can be improved to **always** show recently viewed artworks.

https://github.com/artsy/eigen/assets/11945712/de3ffc8c-da77-443c-b8bf-9305aabb77ea

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add 404 page to artwork screen - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-678]: https://artsyproduct.atlassian.net/browse/ONYX-678?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (add 404 page to artwork screen - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8aceac6271c5bca18f18ffbf8f392e49fe25e35d
</p>

File: src/app/Scenes/Artwork/Components/ArtworkError.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+  return (
+    <Flex flex={1}>
+      <FancyModalHeader onLeftButtonPress={goBack} />
+
+      <ScrollView contentContainerStyle={{ paddingBottom: 80 }}>
+        <Flex p={2}>
+          <Text variant="lg-display">
+            Sorry, the page you were looking for doesn't exist at this URL.
+          </Text>
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/ArtworkError.tsx:36]: **question(non-blocking):** should we add the same copy as force here maybe? https://github.com/artsy/force/pull/13466/files#diff-02987319a18b757067912f4ddf1c305dbd6742a3058e819b82ed27ea7fa54a7cR13-R14

**Review by gkartalis — COMMENTED** [review]
Looks great in general. Left only one minor comment about the copy change.

Not sure if we should **always** display artworks in case of 404s in other pages, but like for example it would be nice if we have a 404 on a fair to show other fairs or 404 on shows display other shows or something.

Def many options but not sure who will take this decision. Thanks for this pr eitherway though 🤩

**Comment by MounirDhahri** [comment]
That's a nice idea @gkartalis - I can look into custom error pages per screen :)

**Review by gkartalis — APPROVED** [review]

**Pull request merged**